### PR TITLE
tests: Add triggering random go failpoints to linearizability tests

### PR DESCRIPTION
**Pull request by serathius**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: tests/linearizability/linearizability_test.go [hunk 1]
```diff
@@ -128,6 +130,7 @@ func triggerFailpoints(ctx context.Context, clus *e2e.EtcdProcessCluster, config
 	for successes < config.count && failures < config.count {
 		err = config.failpoint.Trigger(ctx, clus)
 		if err != nil {
+			fmt.Printf("Failed to trigger failpoint %q, err: %v\n", config.failpoint.Name(), err)
```
Comment (ahrtr) [inline tests/linearizability/linearizability_test.go:133]:
```suggestion
			t.Logf("Failed to trigger failpoint %q, err: %v\n", config.failpoint.Name(), err)
```

File: tests/linearizability/linearizability_test.go [hunk 2]
```diff
@@ -130,7 +130,7 @@ func triggerFailpoints(ctx context.Context, clus *e2e.EtcdProcessCluster, config
 	for successes < config.count && failures < config.count {
 		err = config.failpoint.Trigger(ctx, clus)
 		if err != nil {
-			fmt.Printf("Failed to trigger failpoint %q, err: %v\n", config.failpoint.Name(), err)
+			t.Logf("Failed to trigger failpoint %q, err: %v\n", config.failpoint.Name(), err)
```
Comment (ahrtr) [inline tests/linearizability/linearizability_test.go:133]: You need to pass` t *testing.T` to the function.

**Comment by serathius** [comment]
ping @ahrtr

**Review by ahrtr — APPROVED** [review]
LGTM 

Thanks @serathius

**Pull request merged**